pkg-manager/resolve-dependencies/src/resolvePeers.ts

```diff
@@ -78,16 +81,30 @@ export function resolvePeers<T extends PartialResolvedPackage> (
   const rootPkgsByName = opts.resolvePeersFromWorkspaceRoot ? getRootPkgsByName(opts.dependenciesTree, opts.projects) : {}
   const peerDependencyIssuesByProjects: PeerDependencyIssuesByProjects = {}
 
-  for (const { directNodeIdsByAlias, topParents, rootDir, id } of opts.projects) {
+  for (const { directNodeIdsByAlias, manifest, topParents, rootDir, id } of opts.projects) {
     const peerDependencyIssues: Pick<PeerDependencyIssues, 'bad' | 'missing'> = { bad: {}, missing: {} }
     const pkgsByName = {
       ...rootPkgsByName,
       ..._createPkgsByName({ directNodeIdsByAlias, topParents }),
     }
 
+    const npmAliases: Record<string, string[]> = {}
+    const allDependencies = {
+      ...manifest.dependencies,
+      ...manifest.devDependencies,
+    }
+    for (const [pkgName, version] of Object.entries(allDependencies)) {
+      if (version.startsWith('npm:')) {
+        const aliasedPkgName = version.substring(4).split('@')[0]
+        npmAliases[aliasedPkgName] = npmAliases[aliasedPkgName] ?? []
+        npmAliases[aliasedPkgName].push(pkgName)
+      }
+    }
```

Comment:
The way you did it, aliased packages will be used to resolve peer dependencies only if they are direct dependencies of the project. But aliases should resolve peers from any parent dependencies. See how `parentPkgs` is extended with more dependencies every time we walk the graph deeper.

Reply:
`ctx.parentPkgs` is a map with "aliased name" -> "node", but in `_resolvePeers` we can only get the unaliased name.

```ts
for (const peerName in ctx.resolvedPackage.peerDependencies) { ... }
```